packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,326 @@
+
+		sharedMap1.set("key2", "value2");
+		sharedCell1.set("value2");
+		await provider.ensureSynchronized();
+
+		const opSpaceId1 = getIdCompressor(sharedMap1).normalizeToOpSpace(localId1);
+		const opSpaceId2 = getIdCompressor(sharedCell1).normalizeToOpSpace(localId2);
+		const opSpaceId3 = getIdCompressor(sharedMap1).normalizeToOpSpace(finalId3);
+		const opSpaceId4 = getIdCompressor(sharedCell1).normalizeToOpSpace(finalId4);
+
+		assert(opSpaceId1 >= 0);
+		assert(opSpaceId2 >= 0);
+		assert(opSpaceId3 >= 0 && opSpaceId3 === finalId3);
+		assert(opSpaceId4 >= 0 && opSpaceId4 === finalId4);
+
+		assert.equal(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId1,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			localId1,
+		);
+		assert.equal(
+			getIdCompressor(sharedCell1).normalizeToSessionSpace(
+				opSpaceId2,
+				getIdCompressor(sharedCell1).localSessionId,
+			),
+			localId2,
+		);
+		assert.equal(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId3,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			finalId3,
+		);
+		assert.equal(
+			getIdCompressor(sharedCell1).normalizeToSessionSpace(
+				opSpaceId4,
+				getIdCompressor(sharedCell1).localSessionId,
+			),
+			finalId4,
+		);
+	});
+
+	it("produces Id spaces correctly", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		assert(getIdCompressor(sharedMap2) !== undefined);
+		assert(getIdCompressor(sharedMap3) !== undefined);
+
+		const firstId = getIdCompressor(sharedMap1).generateCompressedId();
+		const secondId = getIdCompressor(sharedMap2).generateCompressedId();
+		const thirdId = getIdCompressor(sharedMap2).generateCompressedId();
+		const decompressedIds: string[] = [];
+
+		const firstDecompressedId = getIdCompressor(sharedMap1).decompress(firstId);
+		decompressedIds.push(firstDecompressedId);
+		sharedMap1.set(firstDecompressedId, "value1");
+
+		[secondId, thirdId].forEach((id, index) => {
+			assert(getIdCompressor(sharedMap2) !== undefined);
+			const decompressedId = getIdCompressor(sharedMap2).decompress(id);
+			decompressedIds.push(decompressedId);
+			sharedMap2.set(decompressedId, `value${index + 2}`);
+		});
+
+		// should be negative
+		assert(getIdCompressor(sharedMap1).normalizeToOpSpace(firstId) < 0);
+		assert(getIdCompressor(sharedMap2).normalizeToOpSpace(secondId) < 0);
+		assert(getIdCompressor(sharedMap2).normalizeToOpSpace(thirdId) < 0);
+
+		await provider.ensureSynchronized();
+
+		assert(getIdCompressor(sharedMap1).normalizeToOpSpace(firstId) > 0);
+		assert(getIdCompressor(sharedMap2).normalizeToOpSpace(secondId) > 0);
+		assert(getIdCompressor(sharedMap2).normalizeToOpSpace(thirdId) > 0);
+
+		decompressedIds.forEach((id, index) => {
+			assert.equal(sharedMap1.get(id), `value${index + 1}`);
+			assert.equal(sharedMap2.get(id), `value${index + 1}`);
+		});
+	});
+
+	// IdCompressor is at container runtime level, which means that individual DDSs
```

Comment:
This also means that individual data stores in the same container should have the same underlying compressor state. You should have tests for those too.

Reply:
I have a test for this scenario now.